modules/transport/Transport.js

```diff
@@ -0,0 +1,191 @@
+import EventEmitter from 'events';
+
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+import PostMessageTransport from './PostMessageTransport';
+
+/**
+ * Stores the currnet transport that have to be used.
+ */
+export default class Transport extends EventEmitter {
```

Comment:
I don't have preference about the name. Although right now it is more of a transport manager or transport facade.